test/fail_compilation/test16228.d

```diff
@@ -0,0 +1,24 @@
+/* REQUIRED_ARGS: -dip25
+   TEST_OUTPUT:
+---
+fail_compilation/test16228.d(22): Error: function has 'return' but does not return any indirections
+fail_compilation/test16228.d(23): Error: function test16228.S.foo non-static member has no 'this' to which 'return' can apply
```

Comment:
I think this should be `Error: function test16228.S.foo static member has no 'this' to which 'return' can apply`, because `foo` is in fact a static member.

To me the following sounds better:
```
Error: function test16228.S.foo is 'static' and can't be have 'return' function attribute which is only applicable to non-static member functions.
```

Or:

```
Error: function test16228.S.foo is ststic. Only instance member functions can have 'return' attribute.
```